Sources/Core/UploadClient/UploadClient.swift

```diff
@@ -2,6 +2,11 @@ import Foundation
 
 /// A type used to upload traces asynchronously during a test run.
 struct UploadClient {
+  enum UploadError: Error {
+    case error(message: String)
+    case unknown
+  }
```

Comment:
```suggestion
  enum UploadError: LocalizedError {
    case error(message: String)
    case unknown

    var errorDescription: String? {
      switch self {
      case .error(let message): return message
      case .unknown: return "Unknown Error"
      }
    }
  }
```
Love this addition.
I think we should also conform to `LocalizedError` for use with `error.localizedDescription`